### ISSUE TITLE
AndroidX support in butterknife gradle plugin

### ISSUE THREAD

**Issue by marcin-kozinski** [JakeWharton/butterknife#1300]
I've tried migrating our project to AndroidX and even using 9.0.0-SNAPSHOT we get compile errors. Looks like the errors are coming from a library module that uses the butterknife gradle plugin.

Here's an example error:
```
butterknifepluginandroidxissue/lib/build/generated/not_namespaced_r_class_sources/debug/generateDebugRFile/out/kozinski/marcin/lib/R2.java:4: error: package android.support.annotation does not exist
import android.support.annotation.AttrRes;
                                 ^
```

Looks like the plugin doesn't check if the project is using AndroidX and always generates an `R2` class with references to `android.support` annotations.

I've uploaded a sample project that has this issue: https://github.com/marcin-kozinski/butterknifepluginandroidxissue
You can go back to the initial commit to check that the project is compiling successfully before the migration to AndroidX.

**Comment by marcin-kozinski** [comment]
Right, sorry about that. Pushed a commit that updates to 9.0.0-SNAPSHOT. The error remains.

**Comment by JakeWharton** [comment]
The Gradle plugin wasn't updated to know about AndroidX and there's no great way to know on way or the other. For now we can use the property, but the property is only required when using jetifier which will eventually go away.

PR welcome for the Gradle plugin support.

**Comment by marcin-kozinski** [comment]
@sharmag222cool Unfortunately no, there is no solution so far. If you have the time, feel free to send a PR that fixes it.

@khatv911 I believe it was a note about how a fix for this could be implemented in the library. So not something you could use as a user of the library. Something that could help you implement a PR to fix it in the library.

**Comment by zerezhka** [comment]
Have the same problem. Butterknife depends on com.android.support:.....25.3.0
list of dependencies 

> (./gradlew :app:dependencies)

![image](https://user-images.githubusercontent.com/10316435/44302425-4885ed00-a339-11e8-8abd-2672dbb7cc09.png)

**Comment by Eipeks** [comment]
I'm having the same issues. 
Generated classes are referencing android support libraries.

Anyone with a workaround for this yet?

**Comment by Eipeks** [comment]
Found a work around. #1322

**Comment by brudaswen** [comment]
@Eipeks You can try my fixed build until my PR is merged to master.

```
buildscript {
    repositories {
        maven { url 'https://jitpack.io' }
    }

    dependencies {
        // XXX classpath 'com.jakewharton:butterknife-gradle-plugin:9.0.0-SNAPSHOT'
        classpath 'com.github.brudaswen.butterknife:butterknife-gradle-plugin:issue_1300-SNAPSHOT'
    }
}
```
